pkg/storage/stores/testutils.go

```diff
@@ -0,0 +1,232 @@
+package stores
+
+import (
+	"context"
+	"sort"
+	"time"
+
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/model/labels"
+
+	"github.com/grafana/loki/pkg/chunkenc"
+	"github.com/grafana/loki/pkg/ingester/client"
+	"github.com/grafana/loki/pkg/logproto"
+	"github.com/grafana/loki/pkg/logql/syntax"
+	"github.com/grafana/loki/pkg/logqlmodel/stats"
+	"github.com/grafana/loki/pkg/storage/chunk"
+	"github.com/grafana/loki/pkg/storage/chunk/cache"
+	chunkclient "github.com/grafana/loki/pkg/storage/chunk/client"
+	"github.com/grafana/loki/pkg/storage/chunk/fetcher"
+	"github.com/grafana/loki/pkg/storage/config"
+	indexstats "github.com/grafana/loki/pkg/storage/stores/index/stats"
+	"github.com/grafana/loki/pkg/util"
+	util_log "github.com/grafana/loki/pkg/util/log"
+)
+
+type MockChunkStore struct {
+	schemas config.SchemaConfig
+	chunks  []chunk.Chunk
+	client  *mockChunkStoreClient
+	f       chunk.RequestChunkFilterer
+}
+
+// mockChunkStore cannot implement both chunk.Store and chunk.Client,
+// since there is a conflict in signature for DeleteChunk method.
+var (
+	_ Store              = &MockChunkStore{}
+	_ chunkclient.Client = &mockChunkStoreClient{}
+)
+
+func NewMockChunkStore(chunkFormat byte, headfmt chunkenc.HeadBlockFmt, streams []*logproto.Stream) *MockChunkStore {
```

Comment:
iiuc this is taken from `pkg/storage/util_test.go`. I was expecting to see that file removed, or do we need both to avoid circular imports?

Reply:
I will double check. Good catch!